### DISCUSSION TITLE
Ability to embed a document/object

### DISCUSSION THREAD

**Discussion by silent6trinity** [Syslifters/sysreptor#41] [General]
When writing reports, I run into situations where I need to embed a file into a finding or a large scope list within the Scope section. Instead of having to manually list every asset out within markdown, it'd be nice to have a way to embed an actual document into the report.

**Comment by aronmolnar** [comment]
What kind of document would this be (e.g. what file type)?

**Reply by silent6trinity** [reply]
> What kind of document would this be (e.g. what file type)?

Typically, i'd be looking to actually embed either another .pdf, or typical office files; (.docx, .xls, etc.)

**Reply by silent6trinity** [reply]
Has there been any work regarding this? I know you have recently pushed out a way to _import_ objects/files - but is there a way to _embed_ files within the report PDF?

**Reply by aronmolnar** [reply]
I haven't yet understood what you want to achieve by *embedding* files in a PDF (PDF is not a container, it's a document).  
Can you provide a sample PDF what this would look like?

**Reply by silent6trinity** [reply]
Often times for tests, (while currently using Microsoft Word for reporting) I will need to embed the agreed upon scope list of IP addresses/domain names into the report as an .xls file. This is typically done by inserting an _object_ into the .docx file, allowing recipients to click the icon to open the file, rather than sending the .xls file as a separate attachment.

Currently, to achieve this when using a .pdf, I need to _edit_ the PDF in Adobe Acrobat _after_ generating the report > comments > attach file. This creates a clickable icon within the PDF that allows the recipient to click the icon and open the embedded file from the report.

I've included an image which depicts what i'm speaking of.

![image](https://github.com/Syslifters/sysreptor/assets/78173918/3ab8735e-5014-4e8e-825f-37c23dbf951e)

**Reply by aronmolnar** [reply]
All clear - tbh I did not know this is possible in PDFs.

This seems to be possible using WeasyPrint.
@MWedl Can you have a look at this (https://doc.courtbouillon.org/weasyprint/stable/api_reference.html#pdf)?

If this works flawlessly, we could allow uploading files to markdown fields (additionally to images). Passing the files to the rendering job could work just as it does with images.

Concerns would be:
* Large PDF file size due to large embedded files
* Users might upload files there and not know they are then embedded to the PDF

**Comment by richardschwabe** [comment]
I think generally an importer would make sense, for importing the usual: nmap, burpsuite/zap, nessus/qualys/openvas, nikto, wpscan and maybe a basic txt file with each line = a list item.

**Reply by aronmolnar** [reply]
Ad importer:
There will be a SysReptor CLI that can be used as an importer.

Ad txt:
This works with the standard list object:

![image](https://docs.sysreptor.com/images/bulk-edit-list.gif)
